fapolicy_analyzer/ui/rules/rules_text_view.py

```diff
@@ -13,27 +13,34 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
-import logging
 
-from fapolicy_analyzer.ui.actions import NotificationType, add_notification
-from fapolicy_analyzer.ui.store import dispatch
-from fapolicy_analyzer.ui.strings import RULES_FILE_READ_ERROR
+import gi
 from fapolicy_analyzer.ui.ui_widget import UIBuilderWidget
 
+gi.require_version("GtkSource", "3.0")
+from gi.repository import GtkSource  # isort: skip
+
 
 class RulesTextView(UIBuilderWidget):
     def __init__(self):
         super().__init__()
         self.__text_view = self.get_object("textView")
         self.__text_view.set_show_line_numbers(True)
 
-    def render_rules(self, rules_path: str):
-        text = ""
-        try:
-            with open(rules_path, "r") as f:
-                text = f.read()
-        except (IOError, FileNotFoundError) as e:
-            logging.error("%s: %s", RULES_FILE_READ_ERROR, e)
-            dispatch(add_notification(RULES_FILE_READ_ERROR, NotificationType.ERROR))
+        # add fapolicyd rules lang file
+        self.manager = GtkSource.LanguageManager.get_default()
+        self.manager.set_search_path(
+            [
+                *self.manager.get_search_path(),
+                "/home/addorschs/fapolicy-analyzer/share/sourceview/language-specs/",
```

Comment:
I'm guessing this is still a WIP, but just in case Ill note this line...

Reply:
lol, yes that was a late Friday push to get done. I'll clean that up today.

Reply:
Fixed up. It now loads the lang and style files from resources.